### DISCUSSION TITLE
Importing CookMate's MyCookbook Format

### DISCUSSION THREAD

**Discussion by dstar4138** [mealie-recipes/mealie#1630] [Feature Request]
**Before submitting this feature request I have**
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
I'd like to request an additional migration format for importing recipies; CookMate's MyCookbook format. I'm submitting this feature request to see how popular a format it may be. The problem is it isn't as clean a format as the others currently implemented; I've attached [1] a partial example of the XML extracted from the MCB file from a family member's export.

There are two formats involved, XML and MCB. MCB is just a ZIP which wraps the XML and a directory called `images` with the full pictures in it. The XML schema is somewhat documented here [1]. In theory it should be similar to the Nextcloud migration already implemented, except that the base format is xml instead of json. 

[1] - [mycookbookrecipes.xml.txt](https://github.com/hay-kot/mealie/files/9503059/mycookbookrecipes.xml.txt)
[2] - http://web.archive.org/web/20210413081805/https://cookmate.blog/my-cookbook-xml-schema/

**Additional Information**
- [x] If the feature is accepted I'm willing to submit a PR to provide this feature

(I haven't taken a deeper look into the code-base but would be willing to dig a bit.)

**Comment by Nero3ooo** [comment]
Maybe someone else is interested in it and while there is no direct import, an relative easy work around for cookmate is described as following:
1. export from cookmate as html
2. open each recepie html you want to import and add _itemscope_  to the line where _itemtype="http://schema.org/Recipe"_ (for me it always was line 6) so it looks like this:
`<div class="recipe" itemscope itemtype="http://schema.org/Recipe">`
3. import into mealie as html / json raw format
4. add images manually

**Reply by Stefan-Carton** [reply]
Just echoing this as thanks, I had 72 recipes to move across. Notepad++ and the "Find in Files" helped to a quick find-replace. Swapped `<div class="recipe" itemtype="http://schema.org/Recipe">` for `<div class="recipe" itemscope itemtype="http://schema.org/Recipe">` and it's been successful.

<img width="841" height="558" alt="image" src="https://github.com/user-attachments/assets/d642c8f8-3cf9-4790-83c9-539d58ac2edc" />

Worked a treat! If there is a way to roll that into the build so replace it in HTML files on import, that'd be golden.

**Reply by om-hb** [reply]
@Nero3ooo @Stefan-Carton How did you export your recipes as HTML? On the CookMate website, I only see the following options:
<img width="663" height="530" alt="grafik" src="https://github.com/user-attachments/assets/d9beb12b-292d-4e4c-8a56-65f7712ae0ea" />
Or did you simply manually save each recipe via the browser?

By the way: I found that [Tandoor](https://github.com/TandoorRecipes/recipes) has an import for MyCookBook export files (.mcb) and Mealie has an import for Tandoor exports. So one could theoretically spin up a Tandoor instance, import the MyCookBook / CookMate export there, export the Tandoor database and import it into Mealie. Might be a bit overkill, however. 😄

**Reply by Stefan-Carton** [reply]
@om-hb On holidays so I can't give a fantastic answer but from memory it was the cookmate Android app which provided the option.

**Reply by Nero3ooo** [reply]
@Stefan-Carton I am glad I could help you :)

@om-hb I only use the app, and I'm not sure if it still works. But the steps to export are like that:

1. Open hamburger menu on the top left corner 
2. Select Im- and Export
3. Click on the arrow on the top right corner
4. Export recipes 
5. Select the recipes you want
6. Click on the arrow on the top right corner
7. Select html
8. Click on export 

The recipes should be on your phone (internal storage -> cookmate backup)

**Reply by om-hb** [reply]
Thank you both for your help.
I had already uninstalled the app, and it didn't occur to me, that the app would have different export options than the website.
Now I installed the app again and would you believe it, there "Export as HTML" was an option.
Thanks again for your swift help and enjoy your holidays @Stefan-Carton! ☺️
